Components/Metrics/PointToSurfaceDistance/itkPointToSurfaceDistanceMetric.h

```diff
@@ -0,0 +1,130 @@
+        public SingleValuedPointSetToPointSetMetric< TFixedPointSet, TMovingPointSet >
+{
+public:
+    using Self = PointToSurfaceDistanceMetric;
+    using Superclass = SingleValuedPointSetToPointSetMetric< TFixedPointSet, TMovingPointSet >;
+    using Pointer = SmartPointer< Self >;
+    using ConstPointer = SmartPointer< const Self >;
+
+    /** Types transferred from the base class */
+    using TransformPointer = typename Superclass::TransformPointer;
+    using TransformParametersType = typename Superclass::TransformParametersType;
+    using TransformJacobianType = typename Superclass::TransformJacobianType;
+    using MeasureType = typename Superclass::MeasureType;
+    using DerivativeType = typename Superclass::DerivativeType;
+    using DerivativeValueType = typename Superclass::DerivativeValueType;
+    using FixedPointSetType = typename Superclass::FixedPointSetType;
+    using MovingPointSetType = typename Superclass::MovingPointSetType;
+    using FixedPointSetConstPointer = typename Superclass::FixedPointSetConstPointer;
+    using MovingPointSetConstPointer = typename Superclass::MovingPointSetConstPointer;
+    using PointIterator = typename Superclass::PointIterator;
+    using PointDataIterator = typename Superclass::PointDataIterator;
+    using InputPointType = typename Superclass::InputPointType;
+    using OutputPointType = typename Superclass::OutputPointType;
+    using CoordRepType = typename OutputPointType::CoordRepType;
+    using VnlVectorType = vnl_vector< CoordRepType >;
+    using NonZeroJacobianIndicesType = typename Superclass::NonZeroJacobianIndicesType;
+
+	   /** Constants for the pointset dimensions. */
+    itkStaticConstMacro( FixedPointSetDimension, unsigned int, TFixedPointSet::PointDimension );
+
+    using ImageType = itk::Image< float, itkGetStaticConstMacro( FixedPointSetDimension )>;
+    using InputImageType = itk::Image< unsigned char,  itkGetStaticConstMacro( FixedPointSetDimension ) >;
+    using RescalerType = itk::RescaleIntensityImageFilter< ImageType, ImageType >;
+    using DtFilterType = itk::DanielssonDistanceMapImageFilter< InputImageType, ImageType, ImageType >;
+    using NegateFilterType = itk::InvertIntensityImageFilter< InputImageType, InputImageType >;
+    using AddImageFilt = itk::AddImageFilter< ImageType, ImageType, ImageType >;
+    using SegReaderType =itk::ImageFileReader< InputImageType >;
+    using DTReaderType = itk::ImageFileReader< ImageType  >;
+    using WriterType = itk::ImageFileWriter< ImageType >;
+    using InterpolatorType = itk::BSplineInterpolateImageFunction< ImageType, float, float >;
+    using ConstDTimage = typename ImageType::ConstPointer;
+	 /** Method for creation through the object factory. */
+    itkNewMacro( Self );//OK
+
+    /** Run-time type information (and related methods). */
+    itkTypeMacro( PointToSurfaceDistanceMetric, SingleValuedPointSetToPointSetMetric );//OK
+    /**  Initialize. */
+    virtual void Initialize() noexcept(false);
```

Comment:
I see you marked some member functions `noexcept(false)`, indicating that they _might_ throw an exception. Why would you do so? When I use `noexcept`, it's usually just to indicate that a function does _not_ throw.

Reply:
Fixed.

Reply:
@gokhangg Thanks for your patience. I still see `noexcept(false)` five times in your proposal. Would you want to remove these as well? Or do they have a special reason?

Reply:
All removed

Reply:
All removed

Reply:
Thanks @gokhangg Nowadays I think it's better to only use `noexcept` when the benefit is really clear (which is clearly the case for move-constructors and move-assignment operators). But other people think otherwise... For your information, I did a little lightning talk about `noexcept` at the conference C++ on Sea, two years ago https://www.youtube.com/watch?v=dVRLp-Rwg0k regarding my [noexcept benchmark](https://github.com/N-Dekker/noexcept_benchmark)